clang/lib/Driver/ToolChains/Clang.cpp

```diff
@@ -7105,10 +7105,15 @@ void Clang::ConstructJob(Compilation &C, const JobAction &JA,
   Args.addOptInFlag(CmdArgs, options::OPT_frelaxed_template_template_args,
                     options::OPT_fno_relaxed_template_template_args);
 
-  // -fsized-deallocation is off by default, as it is an ABI-breaking change for
-  // most platforms.
-  Args.addOptInFlag(CmdArgs, options::OPT_fsized_deallocation,
-                    options::OPT_fno_sized_deallocation);
+  // -fsized-deallocation is on by default in C++14 onwards and otherwise off
+  // by default.
+  if (Arg *A = Args.getLastArg(options::OPT_fsized_deallocation,
```

Comment:
We only need `-fno-sized-deallocation` for CC1Option. When neither option is specified, assume implicit `-fsized-deallocation`.

The code change in Darwin.cpp should be changed to a function that decides whether the default is yes or no.

Reply:
Sorry I may not understand what you mean, can I leave it unchanged as this code is just like `-faligned-allocation`. We can change it later.

Reply:
See my other comment. I think this should be fixed.